src/rule_definition_tools.jl

```diff
@@ -249,7 +261,7 @@ function propagation_expr(Δs, ∂s, _conj = false)
         :($(_∂s[1]) * $(Δs[1]))
     end
 
-    return summed_∂_mul_Δs
+    return :($proj($summed_∂_mul_Δs))
```

Comment:
Do you want me to review this here, or in https://github.com/JuliaDiff/ChainRulesCore.jl/pull/395 ?
I don't mind. but right now the tests are only in this branch not in that one.

(Will review this bit tomorrow either way)

Reply:
Maybe clearer there. Maybe the first question though is whether to pull that out at all, I lean towards "yes" so that this is a not-very-breaking change (i.e. doesn't touch anything not using ProjectTo, and frankly we know all those people) while that one will for sure break some tests.

Reply:
Removed